indico/modules/events/registration/util.py

```diff
@@ -105,6 +105,9 @@ def check_registration_email(regform, email, registration=None, management=False
     user = get_user_by_email(email)
     email_registration = regform.get_registration(email=email)
     user_registration = regform.get_registration(user=user) if user else None
+    email_err = validate_email_verbose(email)
+    if email_err:
+        return dict(status='error', conflict='email-invalid', email_error=email_err)
```

Comment:
Hm, didn't we say something about ignoring this if there's a registration and the user/email did not change (`user and user == registration.user`)